File: pset1/mario.c
/**
	solution to CS50 problem set 1 mario
	Implement a program that prints out a half-pyramid of a specified height
 */


#include <cs50.h>
#include <stdio.h>

/**
    print a symbol n times
    @param n number of times to print
    @param the symbol
*/
void print_symbol(int n, char symbol); 

int main(void)
{
	int height;

	// get the user input of pyramid height
	do
	{
		printf("Height: ");
		height = get_int();
	}
	while (height < 0 || height > 23);
    
    // a loop to print different layers of the pyramid
    for (int i = 0; i < height; i++)
    {
        print_symbol(height - 1 - i, ' ');
        print_symbol(i + 2, '#');
        printf("\n");
    }
    
}

// print a symbol n times 
void print_symbol(int n, char symbol)
{
    for (int i = 0; i < n; i++)
    {
        printf("%c", symbol);
    }
}

File: pset6/mario.py
#!/usr/bin/env python3

def main():
    """
    solution to CS50 problem set 6
    print out a double half-pyramid of a specified height (0, 23]
    """
    height = int(input("Height: "))
    if height <= 0 or height > 23:
        height = int(input("Height: "))
    for i in range(height):
        print((height-1-i)*' '+(i+1)*'#'+"  "+(i+1)*'#'+(height-1-i)*' ')
    
if __name__ == "__main__":
    main()